lnd.go

```diff
@@ -274,19 +274,17 @@ func Main(cfg *Config, lisCfg ListenerCfg, shutdownChan <-chan struct{}) error {
 
 	defer cleanUp()
 
-	serverCreds := credentials.NewTLS(tlsCfg)
-	serverOpts := []grpc.ServerOption{grpc.Creds(serverCreds)}
+	restProxyDest := cfg.RPCListeners[0].String()
+	switch {
+	case strings.Contains(restProxyDest, "0.0.0.0"):
+		restProxyDest = strings.Replace(
```

Comment:
this makes it quite useless for me. We host LND in a container, the ports are not exposed outside to the host, so we need to bind on 0.0.0.0.

Reply:
We should add a comment to this block. This only changes the address the REST proxy uses to dial the gRPC server internally. If it would use `0.0.0.0` it would run into the problem of that address not being contained in the TLS certificate (because that will include concrete IP addresses like `127.0.0.1`). The code as it currently is implemented does **not** restrict the gRPC bind address in any way.

Reply:
> this makes it quite useless for me. We host LND in a container, the ports are not exposed outside to the host, we can't predict the network address of the container, so we need to bind on 0.0.0.0.

Depending on the set up, it would be possible to have something like nginx (or any other reverse proxy really) accept/listen on all interfaces, but then forward to the local lnd node (assuming it's in the same "pod").

Reply:
it is not on same pod. The reverse proxy is in his own container and shared by more than one service.

Reply:
> this makes it quite useless for me. We host LND in a container, the ports are not exposed outside to the host, we can't predict the network address of the container, so we need to bind on 0.0.0.0.

This code is just moved, so no behavior change. Added a comment as @guggero suggested.